### ISSUE TITLE
Issue with dictionary properties

### ISSUE THREAD

**Issue by kadrlica** [kadrlica/pymodel#11]
fermiPy/dmsky#22 identifies an issue with initializing dictionary properties. Here is a minimal test case:

```
from collections import OrderedDict as odict
from pymodeler import Model, Property
class Test(Model):
    _params = odict([('x', Property(dtype=dict,required=True, help='test property')),])

m = Test(x={'hello':'world'})
print(m.x)
```
Of course `m.x` should be set to the dictionary `{'hello':'world'}` but it is still set to `None`.

**Comment by eacharles** [comment]
Ok, I switched the order of attempts in Model.set_attributes() to try mappings first, now this works.

I'll open o PR for that now.

**Comment by kadrlica** [comment]
Closed by #12
